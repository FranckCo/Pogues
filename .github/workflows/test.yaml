name: Pogues Test CI

on:
  push:
    branches:
      - '**'

jobs:
  test-build:
    name: Test & build
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: 20
      - run: yarn
      - run: yarn test:coverage
      - run: yarn build
      - name: Upload coverage artifact
        uses: actions/upload-artifact@v4
        with:
          name: coverage
          path: coverage/
      - name: Upload build artifact
        uses: actions/upload-artifact@v4
        with:
          name: build
          path: dist/

  sonarcloud:
    name: SonarCloud Analysis
    runs-on: ubuntu-latest
    needs: test-build # Ensure SonarCloud job runs after the test-build job
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 0

      - name: Download coverage artifact
        uses: actions/download-artifact@v4
        with:
          name: coverage
          path: coverage

      - name: SonarCloud Scan
        uses: sonarsource/sonarcloud-github-action@v1.13
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
        with:
          args: >
            -Dsonar.pullrequest.key=${{ github.event.pull_request.number }}
            -Dsonar.pullrequest.branch=${{ github.head_ref }}
            -Dsonar.pullrequest.base=${{ github.base_ref }}
