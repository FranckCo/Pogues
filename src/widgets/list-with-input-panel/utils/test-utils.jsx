export function getFakeFields(items, inputName) {
  return {
    name: 'FAKE_NAME.FAKE_NAME',
    length: items.length,
    items: items,
    fields: items.map((it, index) => {
      return {
        index,
        name: `${inputName}[${index}]`,
        listFields: {
          get: indexInner => items[indexInner],
        },
      };
    }),
    map(func) {
      return items.map((it, index) => {
        const field = this.fields[index];
        return func.call(this, field.name, field.index, field.listFields);
      });
    },
  };
}
