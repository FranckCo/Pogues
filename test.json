{"id":"cl_sexe","name":"cl_sexe","label":"Sexe","codes":[{"value":0,"label":"Homme"},{"value":1,"label":"Femme"}]}