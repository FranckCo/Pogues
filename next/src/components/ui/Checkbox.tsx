import { useState } from 'react';

import { FormControl } from '@mui/base/FormControl';
import { Input as MuiInput } from '@mui/base/Input';

import { uuid } from '@/utils/utils';

import Label from './Label';

interface CheckboxProps {
  autoFocus?: boolean;
  label?: string;
  labelPosition?: 'right' | 'top';
  onChange: (newValue: string) => void;
  checked?: boolean;
  disabled?: boolean;
  required?: boolean;
}

export default function Checkbox({
  autoFocus = false,
  label = '',
  labelPosition = 'right',
  onChange,
  disabled = false,
  required = false,
}: Readonly<CheckboxProps>) {
  const [id] = useState(uuid());
  return (
    <FormControl
      disabled={disabled}
      className="grid grid-cols-2 items-center"
      required={required}
    >
      <MuiInput
        id={id}
        type={'checkbox'}
        autoFocus={autoFocus}
        onChange={(e) => onChange(e.target.value)}
        slotProps={{
          input: {
            className:
              'cursor-pointer size-[1.125rem] m-2 text-sm border border-solid bg-white focus-visible:outline-0',
          },
        }}
        aria-label={label}
      />
      {label && labelPosition === 'right' ? (
        <Label>
          <label htmlFor={id}>{label}</label>
        </Label>
      ) : null}
    </FormControl>
  );
}
