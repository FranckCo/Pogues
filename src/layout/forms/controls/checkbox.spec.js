import React from 'react';
import { Field } from 'redux-form';
import Checkbox from './checkbox';
import { decoreFormField } from 'utils/test/test-utils';

describe('Checkbox', () => {
  test('Should have this default template', () => {
    const tree = decoreFormField(<Field component={Checkbox} name="checkbox" label="Label Name" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
