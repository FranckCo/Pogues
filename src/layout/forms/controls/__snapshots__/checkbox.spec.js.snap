// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Should have this default template 1`] = `
<div
  className="form-check-inline"
>
  <label
    className="form-check-label"
    htmlFor="input-checkbox"
  >
    <input
      id="input-checkbox"
      name="checkbox"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      type="checkbox"
      value=""
    />
    Label Name
  </label>
</div>
`;
